lib/rspec/matchers/built_in/change.rb

```diff
@@ -212,6 +216,7 @@ def perform_change(event_proc)
             # returns). We need to cache these values before the `before` value
             # they are based on potentially gets mutated.
             @matches_before = values_match?(@expected_before, actual_before)
+            @nomatches_before = !values_match?(@expected_after, actual_before)
```

Comment:
Naming for this variable is broken.
It says "the value or matcher passed in `to(...)` does **not** match the value before the change".